tactics/equality.ml

```diff
@@ -345,35 +360,35 @@ let find_elim hdcncl lft2rgt dep cls ot =
   in
   let inccl = Option.is_empty cls in
   let env = Proofview.Goal.env gl in
-  (* if (is_global Coqlib.glob_eq hdcncl || *)
-  (*     (is_global Coqlib.glob_jmeq hdcncl && *)
-  (*        jmeq_same_dom env sigma ot)) && not dep *)
-  if (is_global_exists "core.eq.type" hdcncl ||
-      (is_global_exists "core.JMeq.type" hdcncl
-       && jmeq_same_dom env sigma ot)) && not dep
+  let is_eq = is_global_exists "core.eq.type" hdcncl in
+  let is_jmeq = is_global_exists "core.JMeq.type" hdcncl && jmeq_same_dom env sigma ot in
+  if (is_eq || is_jmeq) && not dep
   then
+    let sort = elimination_sort_of_clause cls gl in
     let c =
       match EConstr.kind sigma hdcncl with 
       | Ind (ind_sp,u) ->
-        let pr1 =
-          lookup_eliminator env ind_sp (elimination_sort_of_clause cls gl)
-        in
         begin match lft2rgt, cls with
         | Some true, None
         | Some false, Some _ ->
-          let c1 = destConstRef pr1 in
-          let mp,l = Constant.repr2 (Constant.make1 (Constant.canonical c1)) in
-          let l' = Label.of_id (add_suffix (Label.to_id l) "_r")  in
-          let c1' = Global.constant_of_delta_kn (KerName.make mp l') in
-          begin
+          begin match if is_eq then eq_elimination_ref true sort else None with
```

Comment:
This match-expression works as follows: If it is a rewrite of `core.eq.type` and the elimination scheme has been registered then use it. Otherwise it is a Jmeq or the equality elimination scheme has not been registered, in this case, just use the "old" code.